torch/csrc/jit/ir/alias_analysis.h

```diff
@@ -117,6 +117,23 @@ class AliasDb {
   static bool isMutableType(const Value* v);
   static bool isMutableType(const TypePtr& type);
 
+  /**
+   * Mutation API
+   *
+   * These methods allow you to update AliasDb in-place if you are performing
+   * graph mutation.
+   *
+   * WARNING: They do not perform very many correctness checks, the user is
+   * responsible for making sure they are updating AliasDb correctly.
+   * `Lint()`ing the AliasDb can help with this.
+   */
+  // Copy `existing`s aliasing info to `new_value`, and remove `existing`.
+  void replaceWithNewValue(Value* existing, Value* new_value);
```

Comment:
I put a big warning in the comment,idk . The reason they are named this way is that it's the same as the LLVM alias analysis update API. I suppose we can add a `_` or something but idk if it will deter a sufficiently motivated downstream user.

I believe we still need to friend `MutationRemover` because it manipulates write caches as well.

Reply:
I would maybe put something an internal clarification here, since we don't specify anywhere what updating AliasDb correctly means.